src/cpp/polars/SeriesMask.cpp

```diff
@@ -21,6 +21,32 @@ namespace polars {
         //assert(!arma::any(v > 1));  // Np SeriesMask values may be greater than 1
     };
 
+    SeriesMask SeriesMask::iloc(int from, int to, int step) const {
+
+        if(empty()){
+            return SeriesMask();
+        }
+
+        arma::uvec pos;
+
+        auto effective_from = from;
+        auto effective_to = to;
+
+        if((from < 0) && (to > 0)){
+            effective_from = effective_from  + values().size();
+            pos = arma::regspace<arma::uvec>(effective_from, step, to - 1);
+        }
+        else if(from < 0 && to < 0){
+            effective_from = effective_from + values().size();
+            effective_to = effective_to  + values().size();
+            pos = arma::regspace<arma::uvec>(effective_from, step, effective_to - 1);
+        }
+        else {
+            pos = arma::regspace<arma::uvec>(effective_from,  step, effective_to - 1);
```

Comment:
Can you separate the `pos = arma::regspace<arma::uvec>(effective_from,  step, effective_to - 1);` line from the `effective_x` logic?

You seem to be repeating the line `pos = arma::regspace<arma::uvec>(effective_from,  step, effective_to - 1);` for all cases of the `if...else` - the whole point of separating the logic was to not do that.

Reply:
Position logic now moved outside and if/else just set effective boundaries

Reply:
That seems to be true for `Series` but not for `SeriesMask`

Reply:
That has been amended.